### ISSUE TITLE
[EKS] [request]: Ability to specify "volume type" when provisioning Managed worker nodes from EKS console.

### ISSUE THREAD

**Issue by arpipati** [aws/containers-roadmap#1417]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Add the ability to specify "volume type" for Managed worker nodes, when provisioning Managed nodes from the EKS Console

OR

Enable "gp3" as the default volume type for EKS Managed worker nodes. This is the default volume type when managed nodes are created using "eksctl"

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

Currently, by default, only volumes with "gp2" volume type are provisioned for Managed worker nodes when we create a managed node group from the EKS Console. There is no way to specify volume type parameter.

However, when we provision managed nodes with "eksctl", volumes with "gp3" volume type are provisioned for worker nodes by default.

**Are you currently working around this issue?**
How are you currently solving this problem?
1. Either provision Managed nodes with "eksctl"
2. Manually update the volume type for each worker node.
**Additional context**
Anything else we should know?
Reference:
https://github.com/weaveworks/eksctl/issues/2891

Specifically:
https://github.com/weaveworks/eksctl/issues/2891#issuecomment-762781121

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by eovolkov** [comment]
Any updates?

**Comment by k-weaver** [comment]
Bumping this. I'm currently seeing issues with my managed node group hitting capacity on the AWS side and the error is specifically calling out ```We currently do not have sufficient g4dn.xlarge capacity in zones with support for 'gp2' volumes.```

**Comment by mikestef9** [comment]
See this issue https://github.com/awslabs/amazon-eks-ami/issues/626, gp3 is the default for 1.27+ with the EKS optimized AMI.

**Comment by tomislater** [comment]
@k-weaver you can always as an ad-hoc solution create your own launch configuration with gp3 and use this in your node groups.

**Comment by k-weaver** [comment]
> @k-weaver you can always as an ad-hoc solution create your own launch configuration with gp3 and use this in your node groups.

Yeah, well aware of this but creating a launch template comes with the caveats of having to manage it. An example of this would be updating the AMI. This can get tricky when the entire setup is IaC (i.e. versions in state vs actual) which is why having a fully managed node group is really nice.

 

> See this issue [awslabs/amazon-eks-ami#626](https://github.com/awslabs/amazon-eks-ami/issues/626), gp3 is the default for 1.27+ with the EKS optimized AMI.

I am running 1.27 on all my clusters. I can confirm my gov cloud cluster is not launching instances with GP3 storage when using managed node groups; Same for the commercial side. My managed node groups are launching instances with GP2.

AMI in gov: 
- amazon-eks-gpu-node-1.27-v20240117 (ami-08a5d2780af38e300)
- amazon-eks-node-1.27-v20240117 (ami-0267a9b11e0a48cd4)

Example of a node launched from a managed node group:
![image](https://github.com/aws/containers-roadmap/assets/21695822/3fc36f12-07fb-41f4-a784-9d9080f58a5e)

Volume is gp2:
![image](https://github.com/aws/containers-roadmap/assets/21695822/753f7eb9-7e19-45e7-85a7-7eaa7b4a74a9)

**Comment by tomislater** [comment]
Hmm, you can still use managed groups, in Launch Template you can just edit Storage and that's it 🤔. And then on node group use this Launch Template.
<img width="691" alt="Screenshot 2024-02-06 at 19 18 06" src="https://github.com/aws/containers-roadmap/assets/1127164/59173231-0306-47ef-8ca0-b25a1d03cab8">

**Comment by k-weaver** [comment]
> Hmm, you can still use managed groups, in Launch Template you can just edit Storage and that's it 🤔. And then on node group use this Launch Template.>

Yeah I understand this but as previously mentioned when your using IaC there are complications to launch templates when the time comes to update them.

The issue at hand is GP3 should be the default for EKS managed node groups and this is not the case. Or if there are complications due to compatibility (i.e. local zone) we should have the ability to specify an optional parameter to override the default with gp3 or even io1/io2.

**Comment by mikestef9** [comment]
Ok it turns that PR that enabled gp3 for 1.27 was reverted, because some AWS Local Zones only support gp2. Gathering more info on this.

**Comment by mikestef9** [comment]
The soon to be released AL2023 EKS AMI will be gp3 by default https://github.com/awslabs/amazon-eks-ami/issues/1385

**Comment by nferrario** [comment]
I'd also love to see either gp3 becoming the default or we getting a choice like @k-weaver mentioned for BottleRocket AMIs
